### ISSUE TITLE
pre-commit

### ISSUE THREAD

**Issue by eggplants** [RDFLib/sparqlwrapper#195]
https://github.com/RDFLib/sparqlwrapper/pull/184#discussion_r780823349

I think what is needed: 

- format with black
- lint with flake8
- sort import statements with isort
- (lint with mypy after #193)
- ...

**Comment by eggplants** [comment]
@aucampia Are you going to prepare for pre-commit CI?

**Comment by nicholascar** [comment]
Yes, I think we are fine with pre-commit. Can you please use exactly the same `black` & `flake8` settings are rdflib though? You will see in there that there is a [fixed parameters, version-locked `black`](https://github.com/RDFLib/rdflib/blob/master/black.toml) as well as a [Flak8 with a few settings specified](https://github.com/RDFLib/rdflib/blob/master/.flake8).

**Comment by aucampia** [comment]
@eggplants  I made a PR for adding pre-commit to https://github.com/RDFLib/rdflib and enabled pre-commit CI for it.

If you make a PR for adding it here I will enable it here also.

Regarding flake8, we still need to rationalize our config for https://github.com/RDFLib/rdflib - some work was started on this in https://github.com/RDFLib/rdflib/pull/1442 but we have to finish it. So for now I'm not adding it. You can add it but please try and keep PRs small to make it easier to review, enabling all checks for flake8 at once and fixing all their issues at once is maybe going too far, but we will do our best with reviews.

The regarding mypy, I did not add it to pre-commit as we run this as part of tox and our CI, I'm open to other views here, but since tests are separate anyway I don't see much benefit to having mypy in pre-commmit as opposed to in tox.

**Comment by aucampia** [comment]
For reference, PR on RDFLib/rdflib is:

 - https://github.com/RDFLib/rdflib/pull/1672

**Comment by aucampia** [comment]
One of the main benefits of pre-commit.ci for me is the ability to do `pre-commit.ci autofix` in PRs, so what I put in there is mainly geared towards things that can be automatically fixed, like isort, pycln and black.

**Comment by eggplants** [comment]
@aucampia Could you install pre-commit in sparqlwrapper as well as rdflib?
